cli/lib/nf3.mjs

```diff
@@ -164,6 +164,16 @@ class Nf3 {
     return this.subscribeToIncomingViewingKeys();
   }
 
+  /**
+  Gets the number of unprocessed transactions on the optimist
+  @async
+  */
+
+  async unprocessedTransactionCount() {
```

Comment:
Testing deposit, transfer, withdraw and instant withdrawal end to end is done from the perspective of user. 
Optimist is designed to be run by Proposers, Challengers and LPs and as such a user does not run their own instance of optimist other than wallet/nightfall-client.

The test uses this end point to query if the proposer has any transactions in the mempool before the user sends new transactions because the tests were written to work when re-running them. This meant unprocessed transactions for this user in the optimist mempool should not exist. In real usage, a user will never query a proposer for unprocessed transactions before sending new transactions. So we will have to remove this end point that the user can't use.

An equivalent approach to achieve this would be for the user to repeatedly poll their balance to see if their expected balance from the transactions they performed is equal to the balance their database has from processing Blocks Proposed. If not, they wait longer or propose. An example of this can be seen in `test/ping-pong/user-local/src/index.mjs` and in `test/adversary.test.mjs` in adversary2 branch.

Reply:
Hi, I understand it makes little sense to be in `nf3`, but I wonder if we could just use `axios` to simply call the optimist's endpoint in the test itself? I didn't add that endpoint, it was there already, added by #137 .

Reply:
When optimist is hosted in STAGING and PROD deployments, It will not expose any end point to the world.
This test simulates a non browser desktop application built using an SDK.
Since this application sits in the external world, it won't have access to optimist and will not be able to run against Staging and Prod deployments.

Duncan's architecture diagram would be useful to visualise this. 
![image](https://user-images.githubusercontent.com/36898048/154545039-dfa66150-2ff8-402d-8fd4-ad932d5bb573.png)
Replace the nightfall browser with non browser application.

Reply:
This will be solved in a separate issue #513